core/esmf-native-support/src/main/java/org/eclipse/esmf/nativefeatures/Native.java

```diff
@@ -0,0 +1,446 @@
+         for ( final String targetField : fieldNames ) {
+            if ( field.getName().equals( targetField ) ) {
+               RUNTIME_JNI_ACCESS_SUPPORT.get().register( alwaysTrue(), false, field );
+            }
+         }
+      }
+      return this;
+   }
+
+   public Native registerAllConstructorsForJni() {
+      for ( final Constructor<?> constructor : clazz.getDeclaredConstructors() ) {
+         RUNTIME_JNI_ACCESS_SUPPORT.get().register( alwaysTrue(), false, constructor );
+      }
+      return this;
+   }
+
+   public Native registerConstructorForJni( @Nonnull final Class<?>... args ) {
+      try {
+         RUNTIME_JNI_ACCESS_SUPPORT.get().register( alwaysTrue(), false, clazz.getDeclaredConstructor( args ) );
+      } catch ( final NoSuchMethodException exception ) {
+         throw new RuntimeException( "Could not find constructor in " + clazz + " with args "
+               + Arrays.stream( args ).map( Class::toString ).collect( Collectors.joining( ",", "[", "]" ) ) );
+      }
+      return this;
+   }
+
+   public Native registerAllMethodsForJni() {
+      for ( final Method method : clazz.getDeclaredMethods() ) {
+         RUNTIME_JNI_ACCESS_SUPPORT.get().register( alwaysTrue(), false, method );
+      }
+      return this;
+   }
+
+   public Native registerMethodForJni( final String name, @Nonnull final Class<?>... args ) {
+      try {
+         RUNTIME_JNI_ACCESS_SUPPORT.get().register( alwaysTrue(), false, clazz.getDeclaredMethod( name, args ) );
+      } catch ( final NoSuchMethodException exception ) {
+         throw new RuntimeException( "Could not find method " + name + " in " + clazz + " with args "
+               + Arrays.stream( args ).map( Class::toString ).collect( Collectors.joining( ",", "[", "]" ) ) );
+      }
+      return this;
+   }
+
+   public static void addResource( final String resource ) {
+      RESOURCES_REGISTRY.get().addResources( alwaysTrue(), "\\Q" + resource + "\\E" );
+   }
+
+   public static void addResourcesPattern( final String pattern ) {
+      RESOURCES_REGISTRY.get().addResources( alwaysTrue(), pattern );
+   }
+
+   public static void addResourceBundle( final String name ) {
+      RESOURCES_REGISTRY.get().addResourceBundles( alwaysTrue(), name );
+   }
+
+   public static void addClassBasedResourceBundle( final String name, final String className ) {
+      RESOURCES_REGISTRY.get().addClassBasedResourceBundle( alwaysTrue(), name, className );
+   }
+
+   /**
+    * Returns the class for the given class name.
+    *
+    * @param className the class name
+    * @return the class
+    */
+   public static Class<?> clazz( final String className ) {
+      try {
+         return Class.forName( className );
+      } catch ( final ClassNotFoundException exception ) {
+         throw new RuntimeException( "Could not access for reflection registry", exception );
+      }
+   }
+
+   protected static boolean isLinux() {
+      return new IsLinux().getAsBoolean();
```

Comment:
I am not sure, do we need IsMacOS?

Reply:
We can add it if we need it in the native setup on MacOS. Until now, we've only encountered platform-specific behavior in the JDKs on Windows and Linux that we had to work around.